### PR TITLE
Add v1beta2 to all CRDs and fix some deprecation issues

### LINKED ISSUE #4117: Develop the v1beta2 schemas
As part of https://github.com/strimzi/proposals/blob/master/009-crd-v1-roadmap.md#development we need to develop the v1beta2 resources for each CR kind.

### LINKED ISSUE #4119: DocGenerator should support mutltiple schemas
As part of https://github.com/strimzi/proposals/blob/master/009-crd-v1-roadmap.md#development we need to fix the `DocGenerator` to support generating multiple schemas (I don't think it currently does).

### LINKED ISSUE #4120: Generate v1beta2 resources
As part of https://github.com/strimzi/proposals/blob/master/009-crd-v1-roadmap.md#development we need to change the build in master so that we're generating `v1beta2` resources.

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

This PR is part of the preparation for migration to CRD v1:
* It adds the `v1beta2` version for all resources
* It uses `v1beta2` in all examples and documentation
* It keeps the operator using `v1beta1` / `v1alpha1` by default - this is needed to be able to upgrade to Strimzi 0.22 and convert the CRs to `v1beta2` => the operator will start using `v1beta2` only after 0.22
* It fixes the API and the schema reference to make it clear which fields will be removed from the CRs in `v1beta2`
* It fixes the documentation to use the right APi versions in each resource example
* It changes the documentation from using `kubectl apply` to `kubectl create` or `kubectl replace` because the new CRDs with schema per version are too big for `kubectl apply`
* It changes the system tests to use `kubectl create` / kubectl replace` to deploy the operator
* It generates the CRD v1 files for all resources and extends the structural test to cover them.

This should close #4120, close #4117 and close #4119.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Update CHANGELOG.md

### CODE REVIEW

**Review by alesj — APPROVED** [review]
Lgtm.
Did I miss the document with all the deprecated stuff, or there isn't one yet?
(waiting on Paul to create / deliver one ?)

**Comment by scholzj** [comment]
@alesj The document is here: https://gist.github.com/scholzj/e89e1005109e7b653fdc4101a1e84c10 ... it is not directly part of this PR (although the schema reference should mention that the fields are removed for all of them).

File: systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public static void setClassResources() {
                             resource.getKind(), resource.getMetadata().getName(), resource.getMetadata().getNamespace());
                     operation.inNamespace(resource.getMetadata().getNamespace())
                             .withName(resource.getMetadata().getName())
-                            .withPropagationPolicy(DeletionPropagation.FOREGROUND)
+                            .withPropagationPolicy(DeletionPropagation.BACKGROUND)
```
Comment (tombentley) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java:130]: Why did these have to change?

File: systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java [hunk 2]
```diff
@@ -174,14 +175,14 @@ public void applyClusterOperatorInstallFiles(String namespace) {
...
-            LOGGER.info("Applying configuration file: {}", entry.getKey());
+            LOGGER.info("Creating configuration file: {}", entry.getKey());
             String fileContents = entry.getValue();
             if (Environment.isNamespaceRbacScope()) {
                 fileContents = switchClusterRolesToRoles(fileContents);
 
             }
             clusterOperatorConfigs.push(entry.getKey().getPath());
-            cmdKubeClient().namespace(namespace).applyContent(fileContents);
+            cmdKubeClient().namespace(namespace).replaceContent(fileContents);
```
Comment (tombentley) [inline systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:185]: Is this going to need a change to the Strimzu upgrade documentation?

Comment (scholzj) [reply systemtest/src/test/java/io/strimzi/systemtest/AbstractST.java:185]: This should be already done. The issue is that the Kafka CRD with all 3 schemas is too big for `kubectl apply`. I already went through the docs to change it to `kubectl create` for initial deployments and `kubect replace` for upgrade.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java (hunk 1 above)
Comment (scholzj) [reply systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java:130]: Honestly, I don't know. Nobody was able to tell me why this was used and what was the expectation. What it was doing was that it set a finalizer but did not wait for anything. And the finalizer blocked the deletion of the CRDs later.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 
 ## 0.22.0
 
+* Add `v1beta2` version for all resources. `v1beta2` removes all deprecated fields.
```
Comment (laidan6000) [inline CHANGELOG.md:6]: Maybe best to specify exactly where the new version is added. Do you mean added to the documentation and Schema Reference?

File: documentation/shared/attributes.adoc [hunk 4]
```diff
@@ -131,21 +131,29 @@
...
+:KafkaBridgeApiVersion: kafka.strimzi.io/v1beta2
 
 // API Versions previous
-:KafkaApiVersionPrev: kafka.strimzi.io/v1alpha1
-:KafkaConnectApiVersionPrev: kafka.strimzi.io/v1alpha1
-:KafkaConnectS2IApiVersionPrev: kafka.strimzi.io/v1alpha1
-:KafkaTopicApiVersionPrev: kafka.strimzi.io/v1alpha1
-:KafkaUserApiVersionPrev: kafka.strimzi.io/v1alpha1
-:KafkaMirrorMakerApiVersionPrev: kafka.strimzi.io/v1alpha1
+:KafkaApiVersionPrev: kafka.strimzi.io/v1beta1
```
Comment (laidan6000) [inline documentation/shared/attributes.adoc:147]: Do we also need a previous `ApiVersion` attribute?

File: CHANGELOG.md (hunk 3 above)
Comment (scholzj) [reply CHANGELOG.md:6]: It is overall change for everything.

File: documentation/shared/attributes.adoc (hunk 4 above)
Comment (scholzj) [reply documentation/shared/attributes.adoc:147]: Not sure I follow. This is the previous version.

Comment (laidan6000) [reply documentation/shared/attributes.adoc:147]: You can ignore my comment. 
I thought you might need an `ApiVersionPrev` attribute. However, you are setting previous version attributes for each custom resource.

**Review by tombentley — APPROVED** [review]

**Pull request merged**